File: clients/dotnet-Component-MantaTCP/PTCom.ApplicationBlocks.Messaging/PTCom/ApplicationBlocks/Messaging/BrokerMessage.cs
using System;

namespace PTCom.ApplicationBlocks.Messaging
{

    public class BrokerMessage
    {

        private DeliveryMode deliveryMode;

        private int priority;

        private string messageId;

        private string correlationId;

        private string timestamp;

        private string expiration;

        private string destinationName;

        private string textPayload;

        public BrokerMessage()
        {
            deliveryMode = DeliveryMode.TRANSIENT;
            priority = 4;
            messageId = "";
            correlationId = "";
            timestamp = "";
            expiration = "";
            destinationName = "";
            textPayload = "";
        }

        public DeliveryMode DeliveryMode
        {
            get
            {
                return this.deliveryMode;
            }
            set
            {
                this.deliveryMode = value;
            }
        }

        public int Priority
        {
            get
            {
                return this.priority;
            }
            set
            {
                this.priority = value;
            }
        }

        public string MessageId
        {
            get
            {
                return this.messageId;
            }
            set
            {
                this.messageId = value;
            }
        }

        public string CorrelationId
        {
            get
            {
                return this.correlationId;
            }
            set
            {
                this.correlationId = value;
            }
        }

        public string Timestamp
        {
            get
            {
                return this.timestamp;
            }
            set
            {
                this.timestamp = value;
            }
        }

        public string Expiration
        {
            get
            {
                return this.expiration;
            }
            set
            {
                this.expiration = value;
            }
        }

        public string DestinationName
        {
            get
            {
                return this.destinationName;
            }
            set
            {
                this.destinationName = value;
            }
        }

        public string TextPayload
        {
            get
            {
                return this.textPayload;
            }
            set
            {
                this.textPayload = value;
            }
        }
    }
}

File: clients/java-Component-BrokerTCP/src/pt/com/broker/messaging/BrokerMessage.java
package pt.com.broker.messaging;

import java.util.Date;

import org.caudexorigo.text.DateUtil;
import org.caudexorigo.text.StringUtils;

public class BrokerMessage
{
	public int priority;

	public String messageId;

	public String correlationId;

	public String timestamp;

	public String expiration;

	public String destinationName;

	public String textPayload;

	public BrokerMessage()
	{
	}

	public Date expirationAsDate()
	{
		return convertToDate(expiration);
	}

	public Date timestampAsDate()
	{
		return convertToDate(timestamp);
	}

	public void expirationFromDate(Date date)
	{
		expiration = DateUtil.formatISODate(date);
	}

	private Date convertToDate(String dateString)
	{
		try
		{
			if (StringUtils.isNotBlank(dateString))
			{
				return DateUtil.parseISODate(dateString);
			}
			else
			{
				throw new RuntimeException("Invalid date format");
			}
		}
		catch (Throwable t)
		{
			throw new RuntimeException(t);
		}
	}
}
